slides/12_Remix.html

```diff
@@ -24,6 +24,727 @@
+app/routes/
+    ├── _index.tsx
+    ├── about.tsx
+    ├── pricing.tsx
+    ├── blog._index.tsx
+    ├── blog.$slug.tsx
+    ├── blog.archive.tsx
+    └── blog.authors.tsx
+              </script></code></pre>
+            <p>
+              &rarr; e.g.
+              <span class="code">/about</span> and<span class="code"
+                >/blog/archive</span
+              >
+            </p>
+          </section>
+          <section data-auto-animate>
+            <p>Use the dot delimiter for nested routes</p>
+            <pre><code class="hljs no-highlight" data-trim data-line-numbers="5-8" data-id="code"><script type="text/template">
+app/routes/
+    ├── _index.tsx
+    ├── about.tsx
+    ├── pricing.tsx
+    ├── blog._index.tsx
+    ├── blog.$slug.tsx
+    ├── blog.archive.tsx
+    └── blog.authors.tsx
+              </script></code></pre>
+            <p>&rarr; URLs <span class="code">/blog/…</span></p>
+          </section>
+          <section data-auto-animate>
+            <p>
+              Dynamic route segments (parameters) start with a dollar symbol
+            </p>
+            <pre><code class="hljs no-highlight" data-trim data-line-numbers="6" data-id="code"><script type="text/template">
+app/routes/
+    ├── _index.tsx
+    ├── about.tsx
+    ├── pricing.tsx
+    ├── blog._index.tsx
+    ├── blog.$slug.tsx
+    ├── blog.archive.tsx
+    └── blog.authors.tsx
+              </script></code></pre>
+            <p>
+              &rarr; e.g.
+              <span class="code">/blog/five-great-tips-for-2024</span>
+            </p>
+          </section>
+          <section>
+            See
+            <a
+              href="https://remix.run/docs/en/1.19.3/file-conventions/route-files-v2"
+              >Remix docs</a
+            >
+            for more details.
+          </section>
+        </section>
+        <section>
+          <section data-auto-animate>
+            <h3>Route Module API</h3>
+          </section>
+          <section data-auto-animate>
+            <h3>Route Module API</h3>
+
+            <pre><code class="hljs no-highlight" data-trim data-line-numbers="6" data-id="code"><script type="text/template">
+              app/routes/
+                  ├── _index.tsx
+                  ├── about.tsx
+                  ├── pricing.tsx
+                  ├── blog._index.tsx
+                  ├── blog.$slug.tsx
+                  ├── blog.archive.tsx
+                  └── blog.authors.tsx
+                            </script></code></pre>
+          </section>
+          <section>
+            <p><span class="code">blog.$slug.tsx</span></p>
+            <pre><code class="hljs tsx" data-trim data-line-numbers="|3,10|5-9|1,4" data-id="code"><script type="text/template">
+          import { useParams } from "@remix-run/react";
+
+          export default function BlogPostPage() {
+            const { slug } = useParams();
```

Comment:
**Suggestion:** As it's just a re-export from react router, the same definition applies. Also use the typed variant:

```suggestion
            const { slug } = useParams<'slug'>();
```